integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  EntityMap,
+  InterpretationPluginConfig,
+  Jovo,
+  JovoError,
+  NluData,
+  NluPlugin,
+} from '@jovotech/framework';
+
+import {
+  AnalyzeConversationResult,
+  ConversationAnalysisClient,
+  ConversationAnalysisClientOptionalParams,
+  ConversationPrediction,
+  ConversationTaskParameters,
+  ConversationalTask,
+} from '@azure/ai-language-conversations';
+
+import { AzureKeyCredential, KeyCredential, TokenCredential } from '@azure/core-auth';
+import { v4 as uuidV4 } from 'uuid';
+import { INTENT_NONE, PROJECTKIND_CONVERSATION } from './constants';
+
+export interface MicrosoftCluLibraryConfig {
+  id: string;
+  participantId: string;
+  taskParameters: ConversationTaskParameters;
+  options?: ConversationAnalysisClientOptionalParams;
+}
+
+export interface MicrosoftCluNluConfig extends InterpretationPluginConfig {
+  endpoint: string;
+  credential: TokenCredential | KeyCredential | string;
+  fallbackLanguage: string;
+  libraryConfig: MicrosoftCluLibraryConfig;
+}
+
+export class MicrosoftCluNlu extends NluPlugin<MicrosoftCluNluConfig> {
+  getDefaultConfig(): MicrosoftCluNluConfig {
+    return {
+      ...super.getDefaultConfig(),
+      fallbackLanguage: 'en',
+      endpoint: '',
+      libraryConfig: {
+        id: uuidV4(),
+        participantId: uuidV4(),
+        taskParameters: {
+          projectName: '',
+          deploymentName: '',
+        },
+      },
+    };
+  }
+
+  async processText(jovo: Jovo, text: string): Promise<NluData | undefined> {
+    if (!this.config.endpoint) {
+      throw new JovoError({
+        message: `Can not send request to Microsoft-CLU. The endpoint is missing.`,
+      });
+    }
+
+    if (!this.config.libraryConfig.taskParameters.projectName) {
+      throw new JovoError({
+        message: `Can not send request to Microsoft-CLU. The projectName is missing.`,
+      });
+    }
+
+    if (!this.config.libraryConfig.taskParameters.deploymentName) {
+      throw new JovoError({
+        message: `Can not send request to Microsoft-CLU. The deploymentName is missing.`,
+      });
+    }
+
+    try {
+      const result = await this.sendTextToServer(jovo, text);
+
+      if (
+        result &&
+        result.prediction &&
+        result.prediction.projectKind === PROJECTKIND_CONVERSATION
```

Comment:
We mainly use the short form:
`result?.prediction?.projectKind === PROJECTKIND_CONVERSATION`